libs/langgraph/langgraph/pregel/retry.py

```diff
@@ -63,38 +62,45 @@ def run_with_retry(
                 exc.add_note(f"During task with name '{task.name}' and id '{task.id}'")
             if retry_policy is None:
                 raise
+
+            # Handle single RetryPolicy case
+            if not isinstance(retry_policy, list):
+                retry_policies = [retry_policy]
```

Comment:
this should be done once when creating PregelNode

Reply:
good point!